File: DP/boj_9465.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int num = Integer.parseInt(br.readLine());
		
		for (int i = 0; i < num; i++) {
			int val = Integer.parseInt(br.readLine());
			
			int[][] arr = new int[2][val+1];
			int[][] sticker = new int[2][val+1];

			for (int j = 0; j < 2; j++) {
				StringTokenizer st = new StringTokenizer(br.readLine());
				for (int k = 1; k < val+1; k++) {
					arr[j][k] = Integer.parseInt(st.nextToken());
				}
			}
			sticker[0][1] = arr[0][1];
			sticker[1][1] = arr[1][1];
			for (int k = 2; k < val+1; k++) {
				sticker[0][k] = Math.max(sticker[1][k-1], sticker[1][k-2]) + arr[0][k];
				sticker[1][k] = Math.max(sticker[0][k-1], sticker[0][k-2]) + arr[1][k];
			}
			System.out.println(Math.max(sticker[0][val], sticker[1][val]));
		}
	}	
}

File: DP/boj_9465.py
num = int(input())

for _ in range(num):
    arr = []
    val = int(input())

    for i in range(2):
        arr.append(list(map(int, input().split())))
        
    for j in range(1, val):
        if j == 1:
            arr[0][j] += arr[1][j-1]
            arr[1][j] += arr[0][j-1]
        else:
            arr[0][j] += max(arr[1][j-1], arr[1][j-2])
            arr[1][j] += max(arr[0][j-1], arr[0][j-2])
    
    print(max(arr[0][val-1], arr[1][val-1]))
